classes/conversion.php

```diff
@@ -963,32 +964,47 @@ private function get_transcode_files(\stdClass $conversionrecord, $handler=null)
                 'filepath' => '/' . $conversionrecord->contenthash . '/conversions/',
                 'filename' => $filename,
             );
-
             $downloadparams = array(
                 'Bucket' => $this->config->s3_output_bucket, // Required.
                 'Key' => $availableobject['Key'], // Required.
             );
 
-            $getobject = $s3client->getObject($downloadparams);
-            $filecontent = $getobject->get('Body');
-
             // The playlist files (including iframe playlists) created in s3 transcoding contain
             // relative file paths to Variant Streams for adaptive bitsteaming media, these need to be amended
             // to have Moodle plugin filepaths.
             if ($this->is_file_playlist($filename)) {
+                // This is small enough that we can just pull direct to memory.
+                $result = $s3client->getObject($downloadparams);
+                $filecontent = $result->get('Body');
                 $filecontent = $this->replace_playlist_urls_with_pluginfile_urls($filecontent, $conversionrecord->contenthash);
-            }
 
-            try {
-                $trancodedfile = $fs->create_file_from_string($filerecord, $filecontent);
-            } catch (\moodle_exception $e) {
-                // This file may already exist. Perhaps a reprocessed queue message.
-                // Either way, there isn't anything we can do about it.
-                // Move on.
-                continue;
+                try {
+                    $transcodedfile = $fs->create_file_from_string($filerecord, $filecontent);
+                } catch (\moodle_exception $e) {
+                    // This file may already exist. Perhaps a reprocessed queue message.
+                    // Either way, there isn't anything we can do about it.
+                    // Move on.
+                    continue;
```

Comment:
Instead of having 2 nested try/catch blocks within each if/else block, maybe it's better to move this to outside the if/else block instead?
```
try {
  if (condition) {
    // code...
  } else {
    // code...
  }
} catch () {
  continue;
}